report_wkhtmltopdf_param/models/report_paperformat.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2017 Avoin.Systems
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class Paper(models.Model):
+    _inherit = "report.paperformat"
+
+    custom_params = fields.One2many(
+        "report.paperformat.parameter",
+        "paperformat_id",
+        "Custom Parameters",
+        help="Custom Parameters passed forward as wkhtmltopdf command arguments",
+    )
+
+    @api.constrains("custom_params")
+    def _check_recursion(self):
```

Comment:
```suggestion
    def _has_cycle(self, field_name=None)
```

> Replace the deprecated _check_recursion() method by _has_cycle().

Reply:
change name to `_check_recursion_custom_params`

Reply:
sure, you can do it